components/ai_chat/content/browser/page_content_fetcher.cc

```diff
@@ -256,7 +256,7 @@ class PageContentFetcher {
 
       if (!transcript_text.empty()) {
         // Add a space as a separator betwen texts.
-        transcript_text += " ";
+        transcript_text += ". ";
```

Comment:
Does this possibly lead to double punctuation marks for some transcripts? I see below it is probably needed because of TextEmbedder::SplitSegments thought but maybe it leaves some empty strings in the splits?

Reply:
Yes, it might be depends on the whether the transcript has punctuations or not. Even with double punctuation, those empty segments will be filtered out during split process. But for those without punctuations, we have no way to split them and we can't use space to split because it will generate far too many segments (basically one word per segment)

Reply:
> Even with double punctuation, those empty segments will be filtered out during split process. But for those without punctuations, we have no way to split them and we can't use space to split because it will generate far too many segments (basically one word per segment)

So my concern isn't so much with how this text acts within the context of filtering. But in the cases where it doesn't filter, we'd possibly be sending up to the server with text that has empty sentences.   The LLM is probably smart enough to ignore empty sentences, but it's probably not ideal.

Reply:
I can add this test case to the `TextEmbedderUnitTest.SplitSegments`
```
diff --git a/components/ai_chat/core/browser/text_embedder_unittest.cc b/components/ai_chat/core/browser/text_embedder_unittest.cc
index 9ceff0aeecc..48ff2e43ed1 100644
--- a/components/ai_chat/core/browser/text_embedder_unittest.cc
+++ b/components/ai_chat/core/browser/text_embedder_unittest.cc
@@ -156,6 +156,7 @@ TEST_F(TextEmbedderUnitTest, SplitSegments) {
     std::string input;
     std::vector<std::string> expected;
   } test_cases[] = {{"", {}},
+                    {"HeLoL. . Word", {"HeLoL", "Word"}},
                     {"Hello", {"Hello"}},
                     {"Hello.", {"Hello."}},
                     {"Hello. World!", {"Hello", "World!"}},
```

Reply:
That is removed in https://github.com/brave/brave-core/pull/24953/commits/c84f0d6818dab745a261b0c79c3a9247228f9915.
It might be causing regression even when the feature is disabled